dtfs-central-api/src/v1/controllers/tfm/users/tfm-teams.controller.js

```diff
@@ -1,27 +1,37 @@
+const { validateAuditDetails } = require('@ukef/dtfs2-common/src/helpers/change-stream/validate-audit-details');
+const { generateAuditDatabaseRecordFromAuditDetails } = require('@ukef/dtfs2-common/src/helpers/change-stream/generate-audit-database-record');
 const db = require('../../../../drivers/db-client');
 const { DB_COLLECTIONS, PAYLOAD } = require('../../../../constants');
 const { payloadVerification } = require('../../../../helpers');
 
-const createTeam = async (team) => {
+const createTeam = async (team, auditDetails) => {
   const collection = await db.getCollection(DB_COLLECTIONS.TFM_TEAMS);
-  return collection.insertOne(team);
+  return collection.insertOne({ ...team, auditRecord: generateAuditDatabaseRecordFromAuditDetails(auditDetails) });
 };
 exports.createTeam = createTeam;
 
 exports.createTfmTeam = async (req, res) => {
-  const payload = req?.body?.team;
+  const { team: teamToCreate, auditDetails } = req.body;
 
-  if (payloadVerification(payload, PAYLOAD.TFM.TEAM)) {
-    const team = await createTeam(payload);
+  if (!payloadVerification(teamToCreate, PAYLOAD.TFM.TEAM)) {
+    return res.status(400).send({ status: 400, message: 'Invalid TFM team payload' });
+  }
 
-    const { insertedId } = team;
+  try {
+    validateAuditDetails(auditDetails);
+  } catch ({ message }) {
+    return res.status(400).send({ status: 400, message: `Invalid auditDetails, ${message}` });
+  }
 
-    return res.status(200).json({
-      _id: insertedId,
-    });
+  if (auditDetails.userType !== 'tfm') {
```

Comment:
Note to self: make `userType === 'tfm'` or `'portal'` a reusable helper, and package it up with `validateAuditDetails`